website/content/docs/overview/modules/index.md

```diff
@@ -6,120 +6,60 @@ weight: 40
+![Example](ModuleArch2.png)
 
 
-For each module, the control plane needs **module details** describing the capabilities of the module (read, write, copy, transforms, etc.). This is needed so the control plane can pick the set of modules that can be used together in the same blueprint and fulfill all application and governance requirements. 
+# Components that make up a module
+There are several parts to a module:
+1. [Module Workload](#module-workload): the workload that runs once the Helm chart is installed by the control plane
+2. (Optional) External component: deployed and managed independently of the Mesh for Data.  In this case, the [Module Workload](#module-workload) is a client to the external component.
+3. [Module Helm Chart](#module-helm-chart): the package containing the module workload that the control plane installs as part of a data plane.
+4. [M4DModule YAML](#m4dmodule): describes the functional capabilities, supported interfaces, and has links to the Module Helm chart - that is either the functional component itself or the client to an external component
 
 
-For any module chosen by the control plane to be part of a blueprint, the control plane needs to be able to install/remove/upgrade an instance of the module. For example, if it's a service then the control plane needs to deploy and configure the service. This is done by installing the deployable **module package**.
+# When is a module used?
 
+There are three main data flows in which modules may be used:
+* Read - preparing data to be read and/or actually reading the data
+* Write - writing a new data set or appending data to an existing data set
+* Copy - for performing an implicit data copy on behalf of the application.  The decision to do an implicit copy is made by the control plane, typically for performance or governance reasons.
 
-Once deployed, we have an instance of the module's resources deployed to the cluster. This is the **module code**.
+A module may be used in one or more of these flows, as is indicated in the module's yaml file.
 
+# Credential management
+Modules that access or write data need to receive from the control plane either the credentials to the data stores  or a link to where those credentials reside.  Modules thus can indicate which approach they will use:
+* secret-provider - module calls the Mesh for Data secret provider to obtain the credentials.  TODO: Add link to API
+* automatic - credentials are injected by data-mesh, meaning that the credential values are sent as parameters to the module when the module is called.  
 
-This document describes how {{< name >}} currently implements this concept using `M4DModule` CRD and Helm as well as listing requirements from the module code.
+TODO: It appears that the code currently assumes all modules call the secret provider directly
```

Comment:
Let's keep this section out for now. Originally `secret-provider` meant that the module can fetch from the secret provider while `automatic` meant that it can't so some other module needs to inject the credentials. We don't have an injector module yet because we had stability issues with WASME. If there is no section about credentials in the contributing guide then we need to add something simple that describes the (current) use of secret-provider.

Reply:
I took the section out of the overview and moved it to contributing, leaving instructions about calling the secret provider.